src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadOnlySpanConstructionTest.cs

```diff
@@ -908,5 +927,1328 @@ .maxstack  4
+{";
+            csharp += withCtor ?
+                $@"    public static System.ReadOnlySpan<{type}> StaticData => new System.ReadOnlySpan<{type}>(new {type}[] {{ {args} }});" :
+                $@"    public static System.ReadOnlySpan<{type}> StaticData => new {type}[] {{ {args} }};";
+            csharp += "}";
+
+            var compilation = CreateCompilationWithMscorlibAndSpan(csharp, options: TestOptions.UnsafeReleaseDll);
+            var verifier = CompileAndVerify(compilation, verify: Verification.Skipped);
+            verifier.VerifyIL("Test.StaticData.get", @$"{{
+  // Code size       11 (0xb)
+  .maxstack  1
+  IL_0000:  ldtoken    ""{ldtokenArg}""
+  IL_0005:  call       ""System.ReadOnlySpan<{type}> System.Runtime.CompilerServices.RuntimeHelpers.CreateSpan<{type}>(System.RuntimeFieldHandle)""
+  IL_000a:  ret
+}}");
+        }
+
+        [Fact]
+        public void NonSize1Type_HasCreateSpan_InPlaceConstructionAttempted()
+        {
+            string csharp = RuntimeHelpersCreateSpan + $@"
+
+public class Test
+{{
+    public static int M()
+    {{
+        System.ReadOnlySpan<int> span = new System.ReadOnlySpan<int>(new int[] {{ 1, 2, 3, 4, 5, 6, 7, 8 }});
+        return span[7];
+    }}
+}}";
+            var compilation = CreateCompilationWithMscorlibAndSpan(csharp, options: TestOptions.UnsafeReleaseDll);
+            var verifier = CompileAndVerify(compilation, verify: Verification.Skipped);
+            verifier.VerifyIL("Test.M", @$"
+{{
+  // Code size       21 (0x15)
+  .maxstack  2
+  .locals init (System.ReadOnlySpan<int> V_0) //span
+  IL_0000:  ldtoken    ""<PrivateImplementationDetails>.__StaticArrayInitTypeSize=32 <PrivateImplementationDetails>.8B4B2444E57AED8C2D05A1293255DA1B048C63224317D4666230760935FA4A184""
+  IL_0005:  call       ""System.ReadOnlySpan<int> System.Runtime.CompilerServices.RuntimeHelpers.CreateSpan<int>(System.RuntimeFieldHandle)""
+  IL_000a:  stloc.0
+  IL_000b:  ldloca.s   V_0
+  IL_000d:  ldc.i4.7
+  IL_000e:  call       ""ref readonly int System.ReadOnlySpan<int>.this[int].get""
+  IL_0013:  ldind.i4
+  IL_0014:  ret
+}}
+");
+        }
+
+        [Fact]
+        public void NonSize1Type_HasCreateSpan_MultipleUsesOfSameData_ShareSameBlob()
+        {
+            string csharp = RuntimeHelpersCreateSpan + $@"
+
+public class Test
+{{
+    public static int M()
+    {{
+        System.ReadOnlySpan<int> span1 = new int[] {{ 1, 2, 3, 4, 5, 6, 7, 8 }};
+        int result = span1[1];
+
+        System.ReadOnlySpan<int> span2 = new int[] {{ 9, 10, 11, 12, 13, 14, 15, 16 }};
+        result += span2[2];
+
+        System.ReadOnlySpan<int> span3 = new int[] {{ 1, 2, 3, 4, 5, 6, 7, 8 }};
+        result += span3[3];
+
+        System.ReadOnlySpan<int> span4 = new int[] {{ 9, 10, 11, 12, 13, 14, 15, 16 }};
+        result += span4[4];
+        
+        return result;
+    }}
+}}";
+            var compilation = CreateCompilationWithMscorlibAndSpan(csharp, options: TestOptions.UnsafeReleaseDll);
```

Comment:
>var compilation = CreateCompilationWithMscorlibAndSpan(csharp, options: TestOptions.UnsafeReleaseDll); [](http://example.com/codeflow?start=12&length=102)

Please restore removed empty lines throughout the file and consider keeping the changes only to the necessary minimum

Reply:
Scratch that. I was looking at a partial diff and didn't realize this is a change to a brand ne code